internal/proxy/impl.go

```diff
@@ -2619,6 +2619,20 @@ func (node *Proxy) Flush(ctx context.Context, request *milvuspb.FlushRequest) (*
 
 // Query get the records by primary keys.
 func (node *Proxy) Query(ctx context.Context, request *milvuspb.QueryRequest) (*milvuspb.QueryResults, error) {
+	retrieveRequest := &internalpb.RetrieveRequest{
+		Base: commonpbutil.NewMsgBase(
+			commonpbutil.WithMsgType(commonpb.MsgType_Retrieve),
+			commonpbutil.WithSourceID(paramtable.GetNodeID()),
+		),
+		ReqID: paramtable.GetNodeID(),
+	}
+	return node.QueryImpl(ctx, request, retrieveRequest)
+}
+
+// QueryImpl implements Query
+// can specific the collection id or other params by pass retrieveRequest
+func (node *Proxy) QueryImpl(ctx context.Context, request *milvuspb.QueryRequest, retrieveRequest *internalpb.RetrieveRequest) (*milvuspb.QueryResults, error) {
```

Comment:
make this private, and you don't need to modify `types.go` and mock it

Reply:
mock here to make *MockProxy return QueryImpl resp to test the specific case.